#!/usr/bin/env python
PACKAGE="cabin_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()


#        Name            Type      Level  Description                       Default     Min       Max
gen.add( "Buoyant_Force",   double_t, 1,     "Force of Buoyancy",           398.08,      0,        1000)
gen.add( "Buoyancy_X_POS",  double_t, 1,     "Buoyancy X Pos.",             .027,          -0.5,     0.5)
gen.add( "Buoyancy_Y_POS",  double_t, 1,     "Buoyancy Y Pos.",             .008,         -0.5,     0.5)
gen.add( "Buoyancy_Z_POS",  double_t, 1,     "Buoyancy Z Pos.",             -.036,          -0.5,     0.5)
gen.add( "IS_Submerged",    bool_t,     1,     "Whether is submerged",      False)
                 # PkgName  #NodeName         #Prefix for generated .h include file, e.g. ParametersConfig.py
exit(gen.generate(PACKAGE, "thruster_controller", "VehicleProperties"))
